templateflow/api.py

```diff
@@ -61,6 +61,10 @@ def get(template, raise_empty=False, **kwargs):
     ...
 
     """
+    if TF_LAYOUT is None:
+        from bids import __version__
```

Comment:
If PyBIDS (or components thereof) is not installed, an `ImportError` is raised here.

Reply:
To avoid repeating ourselves, what about:

```Python
def requires_layout(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        if TF_LAYOUT is None:
            from bids import __version__
            raise RuntimeError(f"A layout with PyBIDS <{__version__}> could not be initiated")
        func(*args, **kwargs)
    return wrapper
```

And then just apply:

```Python
@requires_layout
def get(template, raise_empty=False, **kwargs):
    ...

@requires_layout
def templates(**kwargs):
    ...

@requires_layout
def get_metadata(template):
    ...
```

Reply:
neat 👍